pinot-query-runtime/src/test/java/org/apache/pinot/query/mailbox/GrpcSendingMailboxTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.mailbox;
+
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import org.apache.pinot.common.datablock.DataBlock;
+import org.apache.pinot.common.datablock.DataBlockEquals;
+import org.apache.pinot.common.datablock.DataBlockUtils;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.common.datablock.DataBlockBuilder;
+import org.apache.pinot.segment.spi.memory.CompoundDataBuffer;
+import org.apache.pinot.segment.spi.memory.DataBuffer;
+import org.apache.pinot.segment.spi.memory.PinotByteBuffer;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+
+public class GrpcSendingMailboxTest {
+
+  @Test(dataProvider = "byteBuffersDataProvider")
+  public void testByteBuffersToByteStrings(int[] byteBufferSizes, int maxByteStringSize) {
+    List<ByteBuffer> input = Arrays.stream(byteBufferSizes)
+        .mapToObj(this::randomByteBuffer).collect(Collectors.toList());
+    ByteBuffer expected = concatenateBuffers(input);
+
+    List<ByteString> output = GrpcSendingMailbox.toByteStrings(input, maxByteStringSize);
+    for (ByteString chunk: output) {
+      assertTrue(chunk.size() <= maxByteStringSize);
+    }
```

Comment:
Minor suggestion to strengthen the assertion:
```suggestion
    for (ByteString chunk: output.subList(0, output.size() - 1)) {
      assertEquals(chunk.size(), maxByteStringSize);
    }
    assertTrue(output.get(output.size() - 1).size() <= maxByteStringSize);
```